src/AggregateRoot.php

```diff
@@ -78,7 +105,23 @@ private function getAndClearRecordedEvents(): array
 
     private function reconstituteFromEvents(): AggregateRoot
     {
-        $this->getStoredEventRepository()->retrieveAll($this->aggregateUuid)
+        $storedEventRepository = $this->getStoredEventRepository();
```

Comment:
Maybe we should try to find and use a snapshot first. 

In either case we could call `retrieveAllAfterVersion`. If there was no snapshot that function will (or should) start getting all events, if there was a snapshot it should only get events after the snapshot (I guess that already works now)